### DISCUSSION TITLE
RuntimeError: affine matrix of all images should be the same for channel-wise concatenation.

### DISCUSSION THREAD

**Discussion by JaroKuiken** [Project-MONAI/MONAI#8190] [Q&A] [answered]
I have multiple patient folders that each contain a couple hundred 2D DICOM images. These images were taken in two acquisitions immediately after each other (where one acquisition is for odd slices, and the other for even), but together they form one 3D scan volume per patient. I am currently trying to use the LoadImage transform to load these into one 3D volume, however it gives me a runtime error stating that the affines are not the same. The affines that it returns because they are not equal look like this:

1st affine: 
[[ -0.347222   0.         0.       131.1274  ]
 [  0.        -0.347222   0.        39.0027  ]
 [  0.         0.         0.347222 145.481944]
 [  0.         0.         0.         1.      ]]

2nd affine:
[[ -0.347222   0.         0.       131.1274  ]
 [  0.        -0.347222   0.        39.0027  ]
 [  0.         0.         0.347222 145.134722]
 [  0.         0.         0.         1.      ]]

The only difference between these two are the 145.481944 and 145.134722 values. I dont know why these are not the same, and i dont know what to do right now. I dont think i can just make all the affines equal since that would be changing the data, but i also dont understand why they are not the same.

The code im running that causes this is:
`loader = LoadImage(image_only=True)`
`volume = loader(scanPaths)`
Where 'scanPaths' is a sorted list of all the absolute DICOM paths in the folder.

Does anyone know how to help me with this?

**Comment by KumoLiu** [comment]
Hi @JaroKuiken, when combining DICOM images from multiple acquisitions. Since your two sets of images (odd and even slices) are acquired in separate sequences, small variations in the affine matrices are not unusual. In this case, the difference is in the z-component of the third row, which likely reflects a slight offset in slice positioning. This could be due to scanner precision limits or timing differences between the acquisitions.
Perhaps you can load the odd and even slices separately and then adjust the spacing to implement the final concat.
Thanks.

**Reply by JaroKuiken** [reply]
Correction: I apparently attempted it on a dataset that wás taken in one acquisition, but still the affines are not the same. If i try this with the dataset that was taken in different acquisitions, the affines have multiple values that are not equal.

For your suggestion, what do you mean exactly with:

> adjust the spacing to implement the final concat

I tried the same thing with the SimpleITK DicomSeriesReader, and this surprisingly worked without issues. However, as far as i know, there is no built-in way to load this SimpleITK.Image into MONAI or convert it into a MONAI format somehow. Im thinking of writing a custom ImageReader in MONAI that uses the DicomSeriesReader to circumvent this issue. But i am not exactly sure how to do this and im struggling to find it in your documentation, is this possible? And if so, how?

Thanks for your reply so far, hopefully you can help with this too!

**Reply by JaroKuiken** [reply, marked as answer]
I ended up writing the custom ImageReader based on the ITK DicomSeriesReader example here: https://simpleitk.readthedocs.io/en/master/link_DicomSeriesReader_docs.html. This solved my issue!
